akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -5,16 +5,36 @@
 package akka.actor.typed
 
 import scala.annotation.tailrec
-
 import akka.actor.InvalidMessageException
 import akka.actor.typed.internal.BehaviorImpl
 import akka.actor.typed.internal.WrappingBehavior
 import akka.actor.typed.internal.BehaviorImpl.OrElseBehavior
-
 import akka.util.{ LineNumbers, OptionVal }
 import akka.annotation.{ ApiMayChange, DoNotInherit, InternalApi }
 import akka.actor.typed.scaladsl.{ ActorContext => SAC }
 
+import scala.annotation.switch
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object BehaviorTags {
+
+  // optimization - by keeping an identifier for each concrete subtype of behavior
+  // without gaps we can do table switches instead of instance of checks when interpreting
+  // note that these must be compile time constants for it to work
```

Comment:
Optional: In the final commit message or the comment say which benchmark this shows a difference for